src/main/scala/abalone/variant/Variant.scala

```diff
@@ -107,13 +107,19 @@ abstract class Variant private[variant] (
       Some( (category, Move(Piece(situation.player, Role.defaultRole), orig, dest, situation, boardAfter(situation, orig, dest), true, if (category == "pushout") Some(dest) else None)) )
 
     def generateSideMoves(lineOfMarbles: List[Pos], direction: Option[String]): List[(String, Move)] = {
+      // "left" is related to the direction
+      def canLeftSideMove(pos: Pos): Boolean =
+        pos.sideMovesDirsFromDir(direction)._1.fold(false)(p => situation.board.isEmptySquare(Some(p)))
+      def canRightSideMove(pos: Pos): Boolean =
+        pos.sideMovesDirsFromDir(direction)._2.fold(false)(p => situation.board.isEmptySquare(Some(p)))
+
       List(
         if (lineOfMarbles.size == 3) generateSideMoves(List(lineOfMarbles(0), lineOfMarbles(1)), direction)
```

Comment:
I don't understand this line. Why do we trigger a special check in the case we have 3 marbles? Why does it only use the first two? If we are looking at pairs from three, do we need to also check `List(lineOfMarbles(1), lineOfMarbles(2))`?
Also `List(lineOfMarbles(0), lineOfMarbles(1)` == `lineOfMarbles.dropRight(1)` if `lineOfMarbles.size == 3`.

Reply:
for a side move to be possible, all the elements in the line need to be able to move in the direction specified.
we then generate a side move of 3 marbles with pos and 3rdMarblePos.dir(direction)
for side moves of 2 marbles, we generate it with pos and 2ndMarblePos.dir(direction)

Reply:
so if the size is 3, I make a call to itself but with the shorter list of 2 elements to generate side moves of 2 marbles too

Reply:
If we are looking at pairs from three, do we need to also check List(lineOfMarbles(1), lineOfMarbles(2))?
=> no, these side moves will be generated later (or have already been) because it will be triggered from another initial pos

Reply:
I'll change `generateSideMoves(List(lineOfMarbles(0), lineOfMarbles(1)), direction)` to `generateSideMoves(lineOfMarbles.dropRight(1))` thank you

Reply:
Ok I think I understand why we dont need to check the other pair in the list of three.
However, are we not already checking groups of two in another way? (I've not read allll the code to be sure). How would a group of two that is not part of a subgroup of 3 be handled?
Or is the groups of three marbles (`lineOfMarbles`) every group of three Pos?

Reply:
so when we enter the moves generator as (A, B, right) it will generate the side moves of ABC
(and when we enter as C, B, left), the ones for CBA

Reply:
So the way I read it is, if we have A B C as three consecutive marbles of the same colour we generate the side moves of A B as well as A B C. The side moves of B C will be covered when handling C B A later.
What I don't get is when would you cover the side moves of A B if C wasn't a marble of the same colour, and it was either an empty pos or an opponents marble?

Reply:
it is more : 
we will generate the side moves of A B since we know it exists as a valid pair.
Pushing into that same direction of A towards B, if a C exists we will also generate the side moves of A B C